### ISSUE TITLE
Row count not changed after delete entites and flush collection, after server restarted.

### ISSUE THREAD

**Issue by del-zhenwu** [milvus-io/milvus#2531]
**Describe the bug**
```
15:24:34  2020-06-11:07:24:34,3 DEBUG    [client.py:75] Milvus connect failed: 2 times

15:24:44  2020-06-11:07:24:44,6 DEBUG    [client.py:72] Try connect times: 3, 40.01
15:24:44  2020-06-11:07:24:44,17 DEBUG    [client.py:60] milvus-benchmark-test-zw0ivmez.milvus.svc.cluster.local
15:24:44  2020-06-11:07:24:44,17 DEBUG    [client.py:61] 19530
15:24:44  2020-06-11:07:24:44,27 DEBUG    [client.py:72] Try connect times: 0, 0.01
15:24:44  2020-06-11:07:24:44,37 INFO     [k8s_runner.py:774] Loop time: 6
15:24:44  2020-06-11:07:24:44,38 INFO     [k8s_runner.py:781] collection_dnwudvqn
15:24:44  2020-06-11:07:24:44,38 INFO     [k8s_runner.py:782] delete_rand
15:24:44  2020-06-11:07:24:44,46 DEBUG    [client.py:261] (Status(code=0, message='Success!'), 99326)
15:24:44  2020-06-11:07:24:44,54 DEBUG    [client.py:265] Row count: 99326 in collection: <collection_dnwudvqn>
15:24:44  2020-06-11:07:24:44,54 INFO     [client.py:189] collection_dnwudvqn: length to delete: 43
15:24:44  2020-06-11:07:24:44,79 INFO     [client.py:46] Milvus delete run in 0.0s
15:24:44  2020-06-11:07:24:44,205 INFO     [client.py:46] Milvus flush run in 0.13s
15:24:44  2020-06-11:07:24:44,221 DEBUG    [client.py:261] (Status(code=0, message='Success!'), 99326)
15:24:44  2020-06-11:07:24:44,228 DEBUG    [client.py:265] Row count: 99326 in collection: <collection_dnwudvqn>
15:24:44  2020-06-11:07:24:44,229 INFO     [client.py:193] collection_dnwudvqn: count after delete: 99326
15:24:44  2020-06-11:07:24:44,267 DEBUG    [client.py:261] (Status(code=0, message='Success!'), 99326)
15:24:44  2020-06-11:07:24:44,273 DEBUG    [client.py:265] Row count: 99326 in collection: <collection_dnwudvqn>
15:24:44  2020-06-11:07:24:44,274 ERROR    [main.py:70] 
15:24:44  2020-06-11:07:24:44,276 ERROR    [main.py:71] Traceback (most recent call last):
15:24:44    File "main.py", line 68, in queue_worker
15:24:44      runner.run(run_type, collection)
15:24:44    File "/home/jenkins/agent/workspace/milvus-benchmark-0.10.0/milvus_benchmark/k8s_runner.py", line 785, in run
15:24:44      task_run()
15:24:44    File "/home/jenkins/agent/workspace/milvus-benchmark-0.10.0/milvus_benchmark/client.py", line 198, in delete_rand
15:24:44      assert count_before - len(delete_ids) == self.count()
15:24:44  AssertionError
15:24:44  
15:24:44  2020-06-11:07:24:44,276 DEBUG    [k8s_runner.py:68] Start clean up: milvus-benchmark-test-zw0ivmez
```
```
sqlite> select * from TableFiles;
id|table_id|segment_id|engine_type|file_id|file_type|file_size|row_count|updated_time|created_on|date|flush_lsn
1|collection_aooudxf9|1591859243038212000|6|1591859243038212000|7|28998320|54907|1591860135627162|1591859243038214|1200511|0
2|collection_aooudxf9|1591859247323958000|6|1591859247323958000|7|23001680|43614|1591860033233134|1591859247323960|1200511|0
3|collection_aooudxf9|1591859243038212000|6|1591859252264826000|3|2836675|54907|1591860135630350|1591859252264834|1200511|0
4|collection_aooudxf9|1591859247323958000|6|1591859380985575000|3|2277115|43614|1591860033233559|1591859380985576|1200511|0
5|collection_dnwudvqn|1591859384818775000|6|1591859384818775000|7|52000000|99060|1591860148309240|1591859384818782|1200511|0
9|collection_dnwudvqn|1591859384818775000|6|1591859534901112000|3|4983035|99060|1591860148312369|1591859534901115|1200511|0
118|collection_dnwudvqn|1591860124277538000|6|1591860124277538000|4|151840|109|1591860153654177|1591860124277553|1200511|0
122|collection_aooudxf9|1591860142925851000|6|1591860142925851000|4|793000|1525|1591860147016563|1591860142925863|1200511|0
123|collection_aooudxf9|1591860146096317000|6|1591860146096317000|4|16640|32|1591860147015992|1591860146096320|1200511|0
124|collection_aooudxf9|1591860146453054000|6|1591860146453054000|1|809640|1557|1591860147016583|1591860146453067|1200511|0
125|collection_dnwudvqn|1591860152798057000|6|1591860152798057000|4|31200|60|1591860153637409|1591860152798060|1200511|0
126|collection_dnwudvqn|1591860153204251000|6|1591860153204251000|4|87880|169|1591860156305351|1591860153204263|1200511|0
127|collection_dnwudvqn|1591860155358957000|6|1591860155358957000|4|50440|97|1591860156304497|1591860155358959|1200511|0
128|collection_dnwudvqn|1591860155979791000|6|1591860155979791000|1|138320|266|1591860156305378|1591860155979801|1200511|0
129|collection_aooudxf9|1591860156278986000|6|1591860156278986000|1|34840|67|1591860157422822|1591860156278999|1200511|0
130|collection_aooudxf9|1591860157515717000|6|1591860157515717000|5|0|0|1591860157515727|1591860157515727|1200511|0
sqlite> select sum(row_count) from TableFiles where table_id='collection_dnwudvqn';
sum(row_count)
198821
sqlite> select sum(row_count) from TableFiles where table_id='collection_aooudxf9';
sum(row_count)
200223

```
```
[2020-06-11 15:24:03,033][ERROR][ENGINE][HandleException][milvus_server] Encounter exception when clean collection file:database disk image is malformed: database disk image is malformed
[2020-06-11 15:24:03,041][ERROR][ENGINE][GetCollectionFilePath][milvus_server] Collection file doesn't exist: /test/milvus/db_data_8/loop_stability/db/tables/collection_aooudxf9/1591860146453054000/1591860146453054000 in path: /test/milvus/db_data_8/loop_stability/db for collection: collection_aooudxf9
[2020-06-11 15:24:03,041][ERROR][ENGINE][GetCollectionFilePath][milvus_server] Collection file doesn't exist: /test/milvus/db_data_8/loop_stability/db/tables/collection_aooudxf9/1591860156278986000/1591860156278986000 in path: /test/milvus/db_data_8/loop_stability/db for collection: collection_aooudxf9
[2020-06-11 15:24:03,041][ERROR][ENGINE][MergeFiles][milvus_server] Failed to get merge files for collection: collection_aooudxf9 
[2020-06-11 15:24:03,041][ERROR][ENGINE][BackgroundMerge][milvus_server] Failed to get merge files for collection: collection_aooudxf9 reason:Collection file doesn't exist: /test/milvus/db_data_8/loop_stability/db/tables/collection_aooudxf9/1591860156278986000/1591860156278986000
[2020-06-11 15:24:03,056][ERROR][ENGINE][HandleException][milvus_server] Encounter exception when clean collection files:database disk image is malformed: database disk image is malformed
[2020-06-11 15:24:03,060][ERROR][ENGINE][HandleException][milvus_server] Encounter exception when create collection file:database disk image is malformed: database disk image is malformed
[2020-06-11 15:24:03,060][ERROR][ENGINE][CreateCollectionFile][milvus_server] MemTableFile::CreateCollectionFile failed: Database error: Encounter exception when create collection file:database disk image is malformed: database disk image is malformed
[2020-06-11 15:24:03,060][ERROR][ENGINE][Add][milvus_server] [insert][0]MemTableFile::Add: table_file_schema dimension = 0, collection_id =  
[2020-06-11 15:24:03,060][ERROR][ENGINE][Add][milvus_server] [insert][0] Insert failed: Error: Not able to create collection file 
[2020-06-11 15:24:04,074][ERROR][ENGINE][GetCollectionFilePath][wal_thread] Collection file doesn't exist: /test/milvus/db_data_8/loop_stability/db/tables/collection_aooudxf9/1591860146453054000/1591860146453054000 in path: /test/milvus/db_data_8/loop_stability/db for collection: collection_aooudxf9
[2020-06-11 15:24:04,074][ERROR][ENGINE][GetCollectionFilePath][wal_thread] Collection file doesn't exist: /test/milvus/db_data_8/loop_stability/db/tables/collection_aooudxf9/1591860156278986000/1591860156278986000 in path: /test/milvus/db_data_8/loop_stability/db for collection: collection_aooudxf9
[2020-06-11 15:24:04,074][ERROR][ENGINE][ApplyDeletes][wal_thread] Failed to apply deletes: Error: Collection file doesn't exist: /test/milvus/db_data_8/loop_stability/db/tables/collection_aooudxf9/1591860156278986000/1591860156278986000
[2020-06-11 15:24:04,074][ERROR][ENGINE][Flush][wal_thread] Flush collection collection_aooudxf9 failed 
[2020-06-11 15:24:04,086][ERROR][ENGINE][HandleException][milvus_server] Encounter exception when clean collection files:database disk image is malformed: database disk image is malformed
[2020-06-11 15:24:05,093][ERROR][ENGINE][HandleException][milvus_server] Encounter exception when clean collection files:database disk image is malformed: database disk image is malformed
[2020-06-11 15:24:06,090][ERROR][ENGINE][HandleException][milvus_server] Encounter exception when clean collection files:database disk image is malformed: database disk image is malformed
[2020-06-11 15:24:07,095][ERROR][ENGINE][HandleException][milvus_server] Encounter exception when clean collection files:database disk image is malformed: database disk image is malformed

```
**Steps/Code to reproduce behavior**
random choice operations from tasks, including `flush/insert/delete/search/compact/...`

**Expected behavior**

**Environment details**
0.10.0
cpu

collection num: 2
index_type: ivf_pq

**Screenshots**

**Additional context**

**Comment by del-zhenwu** [comment]
The expected result need to define, if the sqlite files is malformed

**Comment by yhmo** [comment]
SQLite is not safe, could be corrupt:
https://www.sqlite.org/howtocorrupt.html

**Comment by tinkerlin** [comment]
We should redefine how Milvus behaves in the event of a metadata server crash or any exception.
@yhmo

**Comment by yhmo** [comment]
I think this issue has been fixed in 0.10.4
